mailet/api/src/main/java/org/apache/mailet/Serializer.java

```diff
@@ -0,0 +1,473 @@
+        public boolean equals(Object other) {
+            return this.getClass() == other.getClass();
+        }
+    }
+
+    Serializer<Float> FLOAT_SERIALIZER = new FloatSerializer();
+
+    class DoubleSerializer implements Serializer<Double> {
+        @Override
+        public JsonNode serialize(Double object) {
+            return DoubleNode.valueOf(object);
+        }
+
+        @Override
+        public Optional<Double> deserialize(JsonNode json) {
+            if (json instanceof NumericNode) {
+                return Optional.of(json.asDouble());
+            } else {
+                return Optional.empty();
+            }
+        }
+
+        @Override
+        public String getName() {
+            return "DoubleSerializer";
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            return this.getClass() == other.getClass();
+        }
+    }
+
+    Serializer<Double> DOUBLE_SERIALIZER = new DoubleSerializer();
+
+    class ArbitrarySerializableSerializer<T extends ArbitrarySerializable<T>> implements Serializer<T> {
+        private static final Logger LOGGER = LoggerFactory.getLogger(ArbitrarySerializableSerializer.class);
+
+        @Override
+        public JsonNode serialize(T serializable) {
+            ArbitrarySerializable.Serializable<T> serialized = serializable.serialize();
+            ObjectNode serializedJson = JsonNodeFactory.instance.objectNode();
+            serializedJson.put("factory", serialized.getFactory().getName());
+            serializedJson.replace("value", serialized.getValue().toJson());
+            return serializedJson;
+        }
+
+        @Override
+        public Optional<T> deserialize(JsonNode json) {
+            return Optional.of(json)
+                    .filter(ObjectNode.class::isInstance)
+                    .map(ObjectNode.class::cast)
+                    .flatMap(this::instantiate);
+        }
+
+        public Optional<T> instantiate(ObjectNode fields) {
+            return Optional.ofNullable(fields.get("factory"))
+                .flatMap(serializer ->
+                    Optional.ofNullable(fields.get("value"))
+                        .flatMap(value -> deserialize(serializer.asText(), AttributeValue.fromJson(value))));
+        }
+
+        @SuppressWarnings("unchecked")
+        private Optional<T> deserialize(String serializer, AttributeValue<?> value) {
+            try {
+                Class<?> factoryClass = Class.forName(serializer);
+                if (ArbitrarySerializable.Factory.class.isAssignableFrom(factoryClass)) {
+                    ArbitrarySerializable.Factory<T> factory = (ArbitrarySerializable.Factory<T>) factoryClass.newInstance();
+                    return factory.deserialize(new ArbitrarySerializable.Serializable<>(value, (Class<ArbitrarySerializable.Factory<T>>) factoryClass));
+                }
+            } catch (Exception e) {
+                LOGGER.error("Error while deserializing", e);
```

Comment:
could helps us more when logging information about `serializer` and attribute `value` also